definitions/tools/filter_sv_vcf_blocklist_bedpe.cwl

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+
+baseCommand: ["/bin/bash", "filter_sv_vcf_blocklist_bedpe.sh"]
+requirements:
+    - class: ResourceRequirement
+      ramMin: 8000
+    - class: DockerRequirement
+      dockerPull: "mgibio/basespace_chromoseq:v12"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: "filter_sv_vcf_blocklist_bedpe.sh"
+        entry: |
+          #!/bin/bash
+          set -eou pipefail
+          set -o errexit
+          
+          INPUT_VCF="$1"
+          BL_BEDPE="$2"
+          SLOPE="$3"
+          OUT_BASE="$4"
+          
+          #BASE=`basename $1`
+          #NAMEROOT=`echo $BASE | perl -pe 's/.vcf(.gz)?$//g'`
+          
+          if [[ "$BL_BEDPE" == 'NONE' ]]; then
+              /usr/local/bin/bedtools sort -header -i "$INPUT_VCF" > $OUT_BASE.vcf 
+              /opt/htslib/bin/bgzip $OUT_BASE.vcf
+              /usr/bin/tabix -p vcf $OUT_BASE.vcf.gz
+          else
+              #CNVkit outputs invalid format like CIPOS=.,894;CIEND=.,894, which can cause svtools vcftobedpe fail
+              if [[ "$INPUT_VCF" =~ \.vcf\.gz$ ]]; then
+                  /bin/zcat "$INPUT_VCF" | /bin/sed -E 's/CIPOS=\.,[0-9]+;CIEND=\.,[0-9]+/CIPOS=0,0;CIEND=0,0/g' > fixed_input.vcf
+              else
+                  /bin/sed -E 's/CIPOS=\.,[0-9]+;CIEND=\.,[0-9]+/CIPOS=0,0;CIEND=0,0/g' "$INPUT_VCF" > fixed_input.vcf
+              fi
+              #svtools vcftobedpe can take either .vcf or .vcf.gz
+              /opt/conda/envs/python2/bin/svtools vcftobedpe -i fixed_input.vcf -o tmp.bedpe
+              /bin/grep '^#' tmp.bedpe > tmp.header
+              /usr/local/bin/bedtools pairtopair -is -slop "$SLOPE" -type notboth -a tmp.bedpe -b "$BL_BEDPE" | /bin/cat tmp.header /dev/stdin | /opt/conda/envs/python2/bin/svtools bedpetovcf -i /dev/stdin | /opt/conda/envs/python2/bin/svtools vcfsort /dev/stdin > "$OUT_BASE.vcf"
+          
+              /opt/htslib/bin/bgzip $OUT_BASE.vcf
+              /usr/bin/tabix -p vcf $OUT_BASE.vcf.gz
+          fi
+
+inputs:
+    input_vcf:
+        type: File
+        inputBinding:
+            position: 1
+        doc: "vcf file to filter"
+    blocklist_bedpe:
+        type: string
```

Comment:
Would this make more sense as a `File` input?  Additionally, it seems like it would make more sense to make it optional rather than expect the string `NONE`.  If there's a reason it can't use the more common way of omitting an input, it'd be nice to have an explanation in the `doc` line about the unique requirement.  (One way to have optional parameters might be to use `getopts` like here: https://github.com/genome/analysis-workflows/blob/c2276ac6374436334e5f0ce3a0197a751c22b1bc/definitions/tools/sequence_align_and_tag.cwl#L29)

Reply:
I agree. I couldn't find a better way to handle optional parameter **without** default for embedded shell and therefore took this compromised approach. Besides `getops`, what do you think this way ?  
https://github.com/genome/analysis-workflows/blob/ece70ac30cd87100a70f7dc64d08fa72724e9416/definitions/tools/docm_gatk_haplotype_caller.cwl#L20
If the optional parameter input is not passed, the positions of inputs behind will be reduced by 1.

Reply:
Sure! Parameter counting works, too.  I suggested `getopts` because it looked like a few other optional arguments already in the tool.  Once multiple arguments could be omitted it's harder to track them solely by counting.  (If they are required but have defaults, it should be safe to remove the `?` from their definitions in the tool CWL.)